lib/workers/schedule_worker/schedules_builder.rb

```diff
@@ -0,0 +1,41 @@
+class ScheduleWorker < WorkerBase
+  class SchedulesBuilder
+    include Vmdb::Logging
+
+    attr_reader :schedules
+
+    def initialize(scheduler, queue, schedule_category = nil)
+      @scheduler = scheduler
+      @queue = queue
```

Comment:
Yes, @imtayadeway, this is the only thing that seemed a bit off.  Maybe the caller could pass in a Proc?

something like this:

### before:

```ruby
    @schedules[:all] << self.system_schedule_every(every, :tags => [:vmdb_appliance_log_config, schedule_category]) do
      @queue.enq :vmdb_appliance_log_config
    end
 ```

### after:
```
    builder.schedule_every(every, :vmdb_appliance_log_config, :tags => [:vmdb_appliance_log_config], Proc.new { @queue.enq :vmdb_appliance_log_config } )